Sources/CacheAdvance/CacheAdvanceError.swift

```diff
@@ -20,7 +20,9 @@ public enum CacheAdvanceError: Error, Equatable {
     case messageLargerThanCacheCapacity
     /// Thrown when a message being appended to a cache that does not overwrite old messages is too large to store in the remaining space.
     case messageLargerThanRemainingCacheSize
+    /// Thrown when the cache file's persisted static header data is inconsistent with the metadata with which the cache was initialized.
+    case fileNotWritable
     /// Thrown when the cache file is of an unexpected format.
-    /// A corrupted file should be deleted. Corruption can occur if an application crashes while writing to the file.
+    /// A corrupted file should be deleted. Corruption can occur if an application crashes while writing to the file, or if a cache was persisted with an incompatible header.
```

Comment:
> Since writes are atomic I thought we could only lose data in some cases

I believe it is possible for the application to crash in the middle of writing to a file handle.

Reply:
interesting.